pkg/controller/constraint/constraint_controller.go

```diff
@@ -42,24 +42,33 @@ const (
 )
 
 type Adder struct {
-	Opa *opa.Client
+	Opa              *opa.Client
+	ConstraintsCache map[string]map[string]bool
```

Comment:
This is a pretty scary type IMO. I can think of two ways to simplify this a bit.
1) The inner map[string]bool is basically a set. You can define a local type (type constraintSet map[string]bool) and then define an Add() and Remove() function for it while you are at it.
2) The outer map has hard-coded keys which are your normalized enforcement action values. Rather than a dynamic map, you could define this as a struct instead, eg:
type ConstraintsCache struct {
  deny map[string]bool
  dryrun map[string]bool
  unrecognized map[string]bool
}
3. You could get crazy and combine both of these into a struct of constraintSets.

Reply:
Actually, disregard this. I just saw Max's comment and he is totally right. This should be a map keyed by the constraint ID/name/whatever that maps to a struct of its tag values.